### PR TITLE
Fix docs for fixed timestep

### PR DESCRIPTION
**Pull request by JoJoJet**
# Objective

The docs for `FixedTime::expend` are unfinished.

### CODE REVIEW

**Review by nicopap — APPROVED** [review]

**Review by nicopap — APPROVED** [review]
LGTM. could be controversial mutably update `FixedTime` in the condition of the while, but it's fine with me.

**Review by devil-ira — APPROVED** [review]

**Review by cart — APPROVED** [review]

File: crates/bevy_time/src/fixed_timestep.rs [hunk 1]
```diff
@@ -106,15 +107,8 @@ pub fn run_fixed_update_schedule(world: &mut World) {
...
-    while check_again {
-        let mut fixed_time = world.resource_mut::<FixedTime>();
-        let fixed_time_run = fixed_time.expend().is_ok();
-        if fixed_time_run {
-            let _ = world.try_run_schedule(FixedUpdate);
-        } else {
-            check_again = false;
-        }
+    while world.resource_mut::<FixedTime>().expend().is_ok() {
+        world.try_run_schedule(FixedUpdate).ok();
```
Comment (cart) [inline crates/bevy_time/src/fixed_timestep.rs:111]: I think `let _ = world.try_run_schedule(FixedUpdate);` is the right way to suppress must_use warnings. `ok()` does "extra work" and adds more unnecessary logical context. It creates questions like "why are we transforming the result into an option if we're just going to ignore it?".

Comment (konsolas) [reply crates/bevy_time/src/fixed_timestep.rs:111]: Maybe we just want `world.run_schedule(FixedUpdate)` which returns `()`?

Comment (JoJoJet) [inline crates/bevy_time/src/fixed_timestep.rs:111]:
```suggestion
        let _ world.try_run_schedule(FixedUpdate);
```

Comment (JoJoJet) [reply crates/bevy_time/src/fixed_timestep.rs]: > Maybe we just want `world.run_schedule(FixedUpdate)` which returns `()`?

That would panic if `FixedUpdate` does not exist. To be honest that might be better than just silently ignoring the error (which we do now), but this PR isn't the right place to change that.

File: crates/bevy_time/src/fixed_timestep.rs [hunk 2]
```diff
@@ -106,15 +107,8 @@ pub fn run_fixed_update_schedule(world: &mut World) {
...
-    while check_again {
-        let mut fixed_time = world.resource_mut::<FixedTime>();
-        let fixed_time_run = fixed_time.expend().is_ok();
-        if fixed_time_run {
-            let _ = world.try_run_schedule(FixedUpdate);
-        } else {
-            check_again = false;
-        }
+    while world.resource_mut::<FixedTime>().expend().is_ok() {
+        let _ world.try_run_schedule(FixedUpdate);
```
Comment (JoJoJet) [inline crates/bevy_time/src/fixed_timestep.rs:111]:
```suggestion
        let _ = world.try_run_schedule(FixedUpdate);
```

**Comment by JoJoJet** [comment]
I'm moving the code changes to a different branch, as they would conflict with the `0.10.2` branch. This PR is now just for docs.

**Review by cart — APPROVED** [review]

**Pull request merged**